typespec-extension/src/code-model-builder.ts

```diff
@@ -211,6 +211,11 @@ export class CodeModelBuilder {
     }
 
     this.sdkContext = createSdkContext(context, "@azure-tools/typespec-java");
+    if (this.supportsAdvancedVersioning()) {
+      this.sdkContext.apiVersion = "all";
+    } else {
+      this.sdkContext.apiVersion = "latest";
+    }
```

Comment:
Don't do this, if not `this.supportsAdvancedVersioning()`.

Also, please read this `sdkContext.apiVersion` into our own `apiVersion` in class (if it is not "all" or "latest")